File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.java
class Solution {
    public void setZeroes(int[][] matrix) {
        int x =1;
        int y =1;
        int m = matrix.length;
        int n = matrix[0].length;
        
        for(int i=0; i<m; i++){
            if(matrix[i][0] == 0) y = 0;
        }
        for(int j=0; j<n; j++){
            if(matrix[0][j] == 0) x = 0;
        }
        
        for(int i=1; i<m; i++){
            for(int j=1; j<n; j++){
                if(matrix[i][j] == 0){
                    matrix[0][j] = 0;
                    matrix[i][0] = 0;
                }
            }
        }
        for(int j=1; j<n; j++){
            if(matrix[0][j] == 0){
                for(int i=1; i<m; i++){
                    matrix[i][j] = 0;
                }
            }
        }
        for(int i=1; i<m; i++){
            if(matrix[i][0] == 0){
                for(int j=1; j<n; j++){
                    matrix[i][j] = 0;
                }
            }
        }
        if(y == 0){
            for(int i=0; i<m; i++){
                matrix[i][0] = 0;
            }
        }
        if(x == 0){
            for(int j=0; j<n; j++){
                matrix[0][j] = 0;
            }
        }
        
    }
}

File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        col0 = 1
        n = len(matrix)
        m = len(matrix[0])
        for i in range(n):
            for j in range(m):
                if(matrix[i][j] == 0):
                    matrix[i][0] = 0
                    
                    if j!=0:
                        matrix[0][j] = 0
                    else:
                        col0 = 0
        
        for i in range(1,n):
            for j in range(1,m):
                if(matrix[i][j] != 0):
                    if (matrix[i][0] == 0 or matrix[0][j] == 0):
                        matrix[i][j] = 0
        
        if matrix[0][0] == 0:
            for j in range(m):
                matrix[0][j] = 0
        
        if col0 == 0:
            for i in range(n):
                matrix[i][0] = 0
        return matrix
        
        